components/blurhash.worker.ts

```diff
@@ -0,0 +1,12 @@
+import { decode, isBlurhashValid } from 'blurhash';
+
+const ctx: Worker = self as any;
```

Comment:
I'm not a fan of this.

Firstly, this is not a component, so it shouldn't live in the Components folder (That's purely for Vue components).

Secondly, this pretty much disable types for anything regarding the worker.

A better way to do this would be to have it live with a web worker-specific tsconfig file, which would enable the `webworker` lib to provide proper context.

It would essentially be [this](https://stackoverflow.com/a/48950463/7617396), but extending the main config and just adding `webworker` in `lib`.

This should make `self` the proper type and give us type checking on everything.

Do note: the tsconfig file will apply to any file in that directory and sub-directories. Hence the need for workers to live alone, somewhere (Maybe as a sub-directory of `plugins` since they should be available globally, not sure)